src/main/java/org/terasology/multiBlock2/system/MultiBlockServerSystem.java

```diff
@@ -111,6 +115,28 @@ public void update(float delta) {
             }
         }
     }
+    @ReceiveEvent
+    public void onStructureSpawn(SendRegionEvent event, EntityRef entity) {
+        ArrayList<Vector3i> blocks = new ArrayList<>();
+        Vector3f location = new Vector3f();
+        for (SpawnBlockRegionsComponent.RegionToFill region : event.regions) {
+            for (Vector3ic pos : region.region) {
+                blocks.add(new Vector3i(pos));
+            }
+        }
+        entity.getComponent(LocationComponent.class).getWorldPosition(location);
+        MultiBlockDefinition definition = new DefaultMultiBlockDefinition("Structure",
+                new Vector3i((int) location.x(), (int) location.y(), (int) location.z()), blocks);
+        Set<EntityRef> multiBlockMainBlockEntities = getMultiBlockMainBlocksInTheWay(definition);
+        if (areAllMultiBlocksInTheWayRelevant(multiBlockMainBlockEntities)) {
+            // Destroy all multi blocks in the way
+            for (EntityRef multiBlockMainBlockEntity : multiBlockMainBlockEntities) {
+                destroyMultiBlock(multiBlockMainBlockEntity);
+            }
+
+            createMultiBlock(definition);
+        }
```

Comment:
For this remaining part, what's the difference to `createMultiBlock(definition)`?

https://github.com/Terasology/MultiBlock/pull/28/files/dda8af8514c2e03f579c620828b816ddf77621d3..844372a3b28a16c295b7eba1298c836152a1d81c#diff-f8602f8f8a58eda2a66e8250dcd9513183b6d88e9793b12fe5ffb4bb30217e3dR346-R375

Reply:
I didn't get the question, but in the remaining part the only difference with the actual implementation is the definition supplied which was made so that it would fit in properly.